src/foundation.rs

```diff
@@ -256,3 +258,75 @@ pub trait NSDate {
 impl NSDate for id {
 
 }
+
+#[repr(C)]
+struct NSFastEnumerationState {
+    pub state: libc::c_ulong,
+    pub items_ptr: *mut id,
+    pub mutations_ptr: *mut libc::c_ulong,
+    pub extra: [libc::c_ulong; 5]
+}
+
+pub struct NSFastIterator {
+    state: NSFastEnumerationState,
+    buffer: [id; 16],
+    mut_val: Option<libc::c_ulong>,
+    len: usize,
+    idx: usize,
+    object: id
+}
+
+impl Iterator for NSFastIterator {
+    type Item = id;
+
+    fn next(&mut self) -> Option<id> {
+        if self.idx >= self.len {
+            self.len = unsafe {
+                msg_send![self.object, countByEnumeratingWithState:&mut self.state objects:self.buffer count:16]
```

Comment:
Should `self.buffer` here be `self.buffer.as_mut_ptr()`? Otherwise, it seems like this would be passing an array of 16 pointers by copy.

Also, I believe a bare int like `16` defaults to `i32`, and this technically should be a `usize`; adding an explicit type somewhere might not hurt.